src/main/c/ssl.c

```diff
@@ -1201,6 +1202,11 @@ TCN_IMPLEMENT_CALL(jlong /* SSL * */, SSL, newSSL)(TCN_STDARGS,
         tcn_ThrowException(e, "cannot create new ssl");
         return 0;
     }
+
+    // Store the handshakeCount in the SSL instance.
+    *handshakeCount = 0;
```

Comment:
@Scottmitch Hmm just leave it as it is for now. Maybe we should create another issue and go over the whole code and check for these. Then when once is detected throw an exception or so. WDYT ?